featurebyte/service/entity_relationship_extractor.py

```diff
@@ -39,72 +39,140 @@ def __init__(
         self.relationship_info_service = relationship_info_service
         self.derive_primary_entity_helper = derive_primary_entity_helper
 
-    async def extract_all_relationships(
-        self,
-        entity_ids: List[ObjectId],
-    ) -> List[EntityRelationshipInfo]:
+    async def _extract_ancestors_or_descendants(
+        self, entity_ids: List[ObjectId], is_ancestor: bool
+    ) -> List[ObjectId]:
         """
-        Extract all relationships of the given entity IDs
+        Extract ancestors or descendants of the given entity IDs
 
         Parameters
         ----------
         entity_ids: List[ObjectId]
             List of entity IDs
+        is_ancestor: bool
+            If True, extract ancestors. Otherwise, extract descendants
 
         Returns
         -------
-        List[EntityRelationshipInfo]
+        List[ObjectId]
         """
-        ancestor_entity_ids = set(entity_ids)
+        output_entity_ids = set()
+        entity_id_key = "_id" if is_ancestor else "ancestor_ids"
         async for entity in self.entity_service.list_documents_iterator(
-            query_filter={"_id": {"$in": list(entity_ids)}}
+            query_filter={entity_id_key: {"$in": entity_ids}}
         ):
-            ancestor_entity_ids.update(entity.ancestor_ids)
+            if is_ancestor:
+                output_entity_ids.update(entity.ancestor_ids)
+            else:
+                output_entity_ids.add(entity.id)
+        return list(output_entity_ids)
+
+    async def _extract_relationship_map(
+        self, anchored_entity_ids: List[ObjectId], to_descendant: bool
+    ) -> Dict[ObjectId, List[EntityRelationshipData]]:
+        """
+        Extract parent-child or child-parent entity relationship map from anchored entity IDs
+        to its ancestors or descendants. If parent_to_child is True, the parent-child relationship map
```

Comment:
Is `parent_to_child` a typo?

Reply:
I rename the variable name. Will revise the docstring here.